packages/ContentCarousel/ContentCarousel.jsx

```diff
@@ -0,0 +1,252 @@
+            targets: direction === 'left' ? decoyRight : decoyLeft,
+            translateX: direction === 'left' ? ['100%', '0%'] : ['-100%', '0%'],
+            duration: 600,
+            complete: () => {
+              switchPages(increment || (direction === 'left' ? 1 : -1))
+            },
+          },
+          0
+        )
+        .add({
+          targets: itemContainer,
+          translateX: '0%',
+          duration: 1,
+        })
+        .add({
+          targets: direction === 'left' ? decoyRight : decoyLeft,
+          opacity: 0,
+          duration: 1,
+        })
+        .add({
+          targets: direction === 'left' ? decoyRight : decoyLeft,
+          translateX: '100%',
+          opacity: 1,
+          duration: 1,
+        })
+    }
+  }
+
+  const handleSlideSkip = increment => {
+    const itemBelt = document.getElementById('itemBelt')
+    const decoyRight = document.getElementById('decoyRight')
+    const decoyLeft = document.getElementById('decoyLeft')
+
+    const mediaQuery = window.matchMedia('(prefers-reduced-motion: reduce)')
+
+    if (mediaQuery.matches) {
+      const tl = anime.timeline({ duration: 400, easing: 'cubicBezier(0.8, 0, 0.55, 0.94)' })
+      tl.add({
+        targets: itemBelt,
+        opacity: 0,
+        complete: () => {
+          switchPages(increment)
+        },
+      }).add({
+        targets: itemBelt,
+        opacity: 1,
+      })
+    } else if (anime.running.length === 0) {
+      if (increment > 1 || increment < -1) {
+        const tl = anime.timeline({ duration: 400, easing: 'cubicBezier(0.8, 0, 0.55, 0.94)' })
+        tl.add({ targets: [decoyRight, decoyLeft], opacity: 0, duration: 1 })
+          .add({
+            targets: itemBelt,
+            filter: 'blur(18px)',
+            translateX: increment > 0 ? '-100%' : '100%',
+            complete: () => {
+              switchPages(increment)
+            },
+          })
+          .add({ targets: itemBelt, translateX: increment > 0 ? '100%' : '-100%', duration: 1 })
+          .add({
+            targets: itemBelt,
+            filter: 'blur(0px)',
+            translateX: '0%',
+          })
+          .add({ targets: [decoyRight, decoyLeft], opacity: 1 })
+      } else {
+        handleSlideTransition(increment < 1 ? 'right' : 'left', increment)
+      }
+    }
+  }
+
+  const handleSwipeGesture = useDrag(
+    ({ dragging, movement, event }) => {
+      if (dragging) {
+        document.body.ontouchmove = e => {
+          e.preventDefault()
+          return false
+        }
+      }
+      if (!dragging && event.type === 'touchend') {
+        document.body.ontouchmove = () => { }
+        if (movement[0] < 0 && currentPage < totalPages) {
+          handleSlideTransition('left', 1)
+        } else if (movement[0] > 0 && currentPage > 1) {
+          handleSlideTransition('right', -1)
+        }
+      }
+    },
+    { filterTaps: true, axis: 'x', threshold: 100 }
+  )
+
+  const preloadImage = url => {
```

Comment:
Out of curiosity, what do the `preloadImage()`, `findImages()`, and the `useEffect()` do?

Reply:
These functions load the `picture` prop of every slide on initial render. It stops images from loading in only on slide switch.